File: Python/SockMerchant.py
def sockMerchant(n, ar):
    # Write your code here
    pairCount={}
    
    for i in range(n):
        if ar[i] not in pairCount.keys():
            pairCount[ar[i]]=False
        else:
            pairCount[ar[i]] = not pairCount[ar[i]]
    return (n-sum(value == False for value in pairCount.values())) // 2 

File: JS-Typescript/SockMerchant.js
function sockMerchant(n, ar) {
    // Write your code here
    let pairCount={}
    
    ar.forEach(function(num){
       if(pairCount[num] === undefined){
           pairCount[num] = false;
       }else{
           pairCount[num] = !pairCount[num]
       }
    });
    
    let count = 0;
    for(var key in pairCount) {
        if(pairCount[key] == false){
            count = count + 1;
        }
    }
    return (n - count)/2
} 